mirrorlist.go

```diff
@@ -10,26 +10,36 @@ import (
 )
 
 func updateMirrorlists() {
+	configReposMutex.RLock()
```

Comment:
I wrote this but I don't like it at all, all the alternatives I had thought about look bad to me.
 - Switching from config.Repos maps to config.Repos syncMaps on config would have caused more harm than good, it doesn't look like [the go yaml package](https://pkg.go.dev/gopkg.in/yaml.v3) would work as expected on those. Also, this is the only for range on which configs are being used.
- The alternative (horrible) approach would be to hard copy the config.Repos map, work on that, then soft copy it back